### ISSUE TITLE
training code

### ISSUE THREAD

**Issue by JunMa11** [yformer/EfficientTAM#1]
thanks for sharing the awesome work.

is there any plan to release the training code?

**Comment by yformer** [comment]
Hi @JunMa11,

Thanks for your interest in our work. 

Our training is compatible to SAM 2, https://github.com/facebookresearch/sam2/tree/main/training 

Let me know if that works for you.

**Comment by zsz00** [comment]
[sam2/tree/main/training](https://github.com/facebookresearch/sam2/tree/main/training)  的 SAM2Train(SAM2Base),  跟EfficientTAM 的SAM2Base  不兼容

**Comment by yformer** [comment]
Hi @zsz00, I adapted the latest SAM 2 codebase for efficient track anything. Can you retry it?

**Comment by zsz00** [comment]
![image](https://github.com/user-attachments/assets/a0132ade-df28-4a60-a8d5-3524c7961b6a)
你更新后有些文件名和模块名的变化, 跟 training/model/sam2.py 不兼容
模块名从sam2 改为efficient_track_anything, 
Sam2.modeling.sam2_base 改为 efficient_track_anything.modeling.efficienttam_utils

**Comment by zsz00** [comment]
通过改动后,可以正常finetune训练
我用MOSE数据 finetune efficienttam_ti_512x512 模型,  在8*A100 上 bs: 8*16   
![image](https://github.com/user-attachments/assets/287fd4f4-9f20-4e93-8884-641075d10e3b)

训练 的log 如下:
![image](https://github.com/user-attachments/assets/955c012f-f041-428d-ac06-6406e1458b1e)

这里的 Losses/train_all_loss 和  Batch Time 都有很大的波动, 这个正常吗? 
你在 train 和 finetune时也会有这么大的波动吗?

**Comment by yformer** [comment]
@zsz00, how many iterations for your warmup?

**Comment by zsz00** [comment]
no warmup,  我沿用的sam2的training 的配置, 里边没有warmup.

**Comment by yformer** [comment]
@zsz00, can you try decreasing the learning rate to see the difference?

**Comment by zsz00** [comment]
@yformer  我把 lr降一个数量级后 还是波动很大, 看不到loss明显下降趋势
![image](https://github.com/user-attachments/assets/1047d62b-6005-491f-bf15-7e0a4252b5fa)
褐色的曲线是 lr 降低的

你能发下你们训练的训练代码和 training config yaml 吗? 我对比下有什么差异.

**Comment by zsz00** [comment]
https://github.com/chayryali/tam/issues/1#issuecomment-2585131762  
This repo is just a benchmark of "Efficient"TAM, finding that the paper's results are not-reproducible from their code and speedups are negligible.
@yformer 这是真的吗?

**Comment by yformer** [comment]
> [chayryali/tam#1 (comment)](https://github.com/chayryali/tam/issues/1#issuecomment-2585131762) This repo is just a benchmark of "Efficient"TAM, finding that the paper's results are not-reproducible from their code and speedups are negligible. [@yformer](https://github.com/yformer) 这是真的吗?

@zsz00, We benchmarked models on A100 using [the benchmark script](https://github.com/yformer/EfficientTAM/blob/main/efficient_track_anything/benchmark.py) w.r.t. FPS,
```
efficienttam_s: 100.4 FPS
efficienttam_s_1: 106.4 FPS
efficienttam_s_2: 124.5 FPS
efficienttam_s_512x512: 270.9 FPS
```
I think our EfficientTAM significantly improved the efficiency. Let me know if you have issues for benchmarking our EfficientTAM with the provided benchmark script.

**Comment by yformer** [comment]
> [@yformer](https://github.com/yformer) 我把 lr降一个数量级后 还是波动很大, 看不到loss明显下降趋势 ![image](https://github.com/user-attachments/assets/1047d62b-6005-491f-bf15-7e0a4252b5fa) 褐色的曲线是 lr 降低的
> 
> 你能发下你们训练的训练代码和 training config yaml 吗? 我对比下有什么差异.

@zsz00, we based on our internal infra for training our models. I did not have a chance to test the [training code](https://github.com/facebookresearch/sam2/tree/main/training) from SAM2. I have not finetuned our model for downstream task.  We train our EfficientTAM on SA-V dataset and a 10% subset of SA-1B. We used a batch size of 256, and a initial learning rate of 6e− 5 for image encoder and 3e−4 for other components of the model. You can see the experiment section 4.1 of [our paper](https://arxiv.org/pdf/2411.18933) for more details. Let me know if you have other questions.

**Comment by JasonShen-SH** [comment]
@zsz00  Hi , 可以share下如何fine-tuning EfficientTAM的 code吗，多谢！

**Comment by JasonShen-SH** [comment]
> [@zsz00](https://github.com/zsz00) Hi , 可以share下如何fine-tuning EfficientTAM的 code吗，多谢！

至于MOSE微调，似乎这是个常见的现象，你可以试试在SAM2上直接微调MOSE，loss收敛同样也很不平稳，这是数据集的复杂特性导致的
